src/network.c

```diff
@@ -227,6 +227,7 @@ int network_send_msg(struct network *net, struct network_msg *nmsg)
 	nonce.type = NONCE_NETWORK;
 	memcpy(&nonce.ctl_ttl, (void *)nmsg + 1, 6);
 	nonce.iv_index = cpu_to_be32(net->iv_index);
+	err1 = network_cache_add(net, nmsg);
 
```

Comment:
It looks good, but you don't need to introduce a new err1 variable here, neither check returned error.

Reply:
Updated...